src/js/components/Carousel/Carousel.js

```diff
@@ -1,242 +1,260 @@
 import React, {
   Children,
-  useCallback,
+  useRef,
   useContext,
-  useEffect,
   useState,
-  useRef,
+  useEffect,
 } from 'react';
-
-import { normalizeColor } from '../../utils';
-import { defaultProps } from '../../default-props';
+import { CarouselChild } from './CarouselChild';
+import { CarouselControls } from './CarouselControls';
+import {
+  StyledCarouselContainer,
+  StyledCarouselInnerContainer,
+} from './StyledCarousel';
+import { CarouselPropTypes } from './propTypes';
+import { Keyboard } from '../Keyboard';
 import { ThemeContext } from '../../contexts';
+import { defaultProps } from '../../default-props';
 
-import { Box } from '../Box';
-import { Button } from '../Button';
-import { Keyboard } from '../Keyboard';
-import { Stack } from '../Stack';
+const handleSelectorNavigation =
```

Comment:
What's the rationale for moving these functions outside of the component? It feels to me like there is almost twice as much code with the current approach, given the number of arguments being called with.

Reply:
These are just closures to try to DRY things up. I think this makes the code cleaner rather than just inlining these fairly complex functions. Ideally they could live in a utility file and just be imported into the functional component but I don't see that pattern used anywhere else - so I decided to just leave them above.

I think closures help with the readability and performance of these functions with so many arguments, but I'd be interested to see if you had other ideas. The only "extra" code it adds (from what I can see) is just instantiating the first part of the closure in the component. I'm open to exploring other ideas as well so if you have any recommendations please let me know!

Reply:
So, if I look at `handleOnNext`:
- the actual code is lines 61-69 (9 lines)
- the function parameters are defined in lines 53-60 (8 lines)
- calling this function to set up `onNext` is done in lines 138-145 (8 lines)
If instead we just put the actual code directly where `onNext` is defined at line 138, we'd still only have a single copy of the code (DRY) but we be able to omit the two 8 line parameter definition blocks. So, less code.
Furthermore, we should probably be using `useCallback` to ensure we don't re-define the function on each render.

Reply:
Makes sense! Addressed this